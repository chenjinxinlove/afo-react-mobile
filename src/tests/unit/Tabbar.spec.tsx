import * as Enzyme from 'enzyme';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import DemoTabBar from '../../pageages/tabbar/demo';
import TabBar from '../../pageages/tabbar/index'

describe('TabBar', () => {
  // 测试是否能正确的显示内容
  it ('should render correct contents', () => {
    const wrapper = Enzyme.shallow(
      <TabBar>
        <TabBar.Item icon="mall">
          国务
        </TabBar.Item>
        <TabBar.Item icon="credit-card" dot={true}>
          订单
        </TabBar.Item>
      </TabBar>
    )
    expect(wrapper.find('.afo-tabbar').children()).toHaveLength(2)
  })
  // 测试卷快照
  it ('test snapshots', () => {
    const component = renderer.create(
      <DemoTabBar />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})