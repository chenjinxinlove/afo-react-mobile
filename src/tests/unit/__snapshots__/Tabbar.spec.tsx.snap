// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBar test snapshots 1`] = `
<div>
  <div
    className="afo-tabbar afo-tabbar--fixed"
    style={
      Object {
        "background": "white",
        "zIndex": 1,
      }
    }
  >
    <div
      className="afo-tabbar-item afo-tabbar-item--active"
      onClick={[Function]}
    >
      <div
        className="afo-tabbar-item__icon"
      >
        <img
          alt=""
          src="https://img.yzcdn.cn/public_files/2017/10/13/793c77793db8641c4c325b7f25bf130d.png"
        />
        
      </div>
      首页
    </div>
    <div
      className="afo-tabbar-item"
      onClick={[Function]}
    >
      <div
        className="afo-tabbar-item__icon"
      >
        <i
          className="afo-icon afo-icon-mall"
          onClick={undefined}
          style={
            Object {
              "color": undefined,
              "fontSize": "21px",
            }
          }
        />
        
      </div>
      国务
    </div>
    <div
      className="afo-tabbar-item"
      onClick={[Function]}
    >
      <div
        className="afo-tabbar-item__icon icon--dot"
      >
        <i
          className="afo-icon afo-icon-credit-card"
          onClick={undefined}
          style={
            Object {
              "color": undefined,
              "fontSize": "21px",
            }
          }
        />
        
      </div>
      订单
    </div>
    <div
      className="afo-tabbar-item"
      onClick={[Function]}
    >
      <div
        className="afo-tabbar-item__icon"
      >
        <i
          className="afo-icon afo-icon-person"
          onClick={undefined}
          style={
            Object {
              "color": undefined,
              "fontSize": "21px",
            }
          }
        />
        <div
          className="icon__info"
        >
          12
        </div>
      </div>
      个人中心
    </div>
  </div>
</div>
`;
